elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/TableContext.java

```diff
@@ -0,0 +1,268 @@
+        // request arguments.
+        columnArgs.putAll(callingColumnArgs);
+
+        // Build a new Context for resolving this column
+        TableContext newCtx = TableContext.builder()
+                        .queryable(queryable)
+                        .alias(tableCtx.getAlias())
+                        .joins(tableCtx.getJoins())
+                        .metaDataStore(tableCtx.getMetaDataStore())
+                        .build();
+        newCtx.putAll(tableCtx);
+
+        newCtx.putAll(prepareArgumentsMap(tableArgs, TBL_PREFIX));
+        newCtx.putAll(prepareArgumentsMap(columnArgs, COL_PREFIX));
+
+        try {
+            Template template = handlebars.compileInline(columnExpr);
+            return template.apply(newCtx);
+        } catch (IOException e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Object resolveSQLHandlebar(final Object context, final Options options)
+                    throws UnsupportedEncodingException {
+        String from = options.hash("from");
+        String column = options.hash("column");
+        int argsIndex = column.indexOf('[');
+        String invokedColumnName = column;
+
+        TableContext currentTableCtx = (TableContext) context;
+        // 'from' is optional, so if not provided use the same table context.
+        TableContext invokedTableCtx = StringUtils.isEmpty(from) ? currentTableCtx
+                                                                 : (TableContext) currentTableCtx.get(from);
+
+        // Physical References starts with $
+        if (invokedColumnName.lastIndexOf('$') == 0) {
+            return invokedTableCtx.get(invokedColumnName);
+        }
+
+        Map<String, Object> pinnedArgs = new HashMap<>();
+        if (argsIndex >= 0) {
+            parseArguments(column.substring(argsIndex), pinnedArgs);
+            invokedColumnName = column.substring(0, argsIndex);
+        }
+
+        Queryable invokedQueryable = invokedTableCtx.getQueryable();
+
+        // Assumption: sql helper will be part of SQLTable only, so table won't be null.
+        SQLTable table = (SQLTable) metaDataStore.getTable(invokedQueryable.getName(), invokedQueryable.getVersion());
+        String invokedColumnExpr = table.getColumnMap().get(invokedColumnName).getExpression();
+
+        Map<String, Object> currentColumnArgs = (Map<String, Object>)
+                        ((Map<String, Object>) currentTableCtx.get(COL_PREFIX))
+                        .get(ARGS_KEY);
+
+        // Ideally pinned arguments wont clash with column arguments
+        // But if it happens, column arguments gets preference so overriding pinned arguments.
+        pinnedArgs.putAll(currentColumnArgs);
+
+        return resolveHandlebars(invokedTableCtx, invokedColumnName, invokedColumnExpr, pinnedArgs);
+    }
+
+    public void addJoin(String joinName, Queryable joinQueryable) {
```

Comment:
We can remove this function if we leverage the joins stored in the associated Queryable.